scripts/new-version-pr-generator.js

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable no-console */
+
+const {execSync} = require('child_process');
+const {readFileSync} = require('fs');
+const pathResolve = require('path').resolve;
+const {mkdir, rm} = require('shelljs');
+const yaml = require('js-yaml');
+const semver = require('semver');
+
+const {version: PACKAGE_VERSION} = require('../package.json');
+const secrets = require('../secrets.json');
+const Retry = require('./utilities/retry');
+
+const YARN_VERSION = yaml
+  .safeLoad(readFileSync(pathResolve(__dirname, '..', 'dev.yml'), 'utf8'))
+  .up.find((obj) => Object.keys(obj).includes('node')).node.yarn;
+
+// File paths and directories
+const root = pathResolve(__dirname, '../');
+const sandbox = pathResolve(root, 'sandbox');
+
+// Create the sandbox directory
+rm('-rf', sandbox);
+mkdir(sandbox);
+
+// Current release version
+const releaseVersion = `v${PACKAGE_VERSION}`;
+
+// Compute the base branch on the repository (default: master)
+// Example: will open a PR against the v4 branch if the
+// version found in package.json is `4.0.0-beta.x`
+
+// Check if a version is a major pre-release:
+// isMajorPrerelease('2.0.0') => false
+// isMajorPrerelease('3.0.1-beta') => false
+// isMajorPrerelease('1.1.0-alpha') => false
+// isMajorPrerelease('2.0.0-beta.1') => true
+// isMajorPrerelease('3.0.0-alpha') => true
+function isMajorPrerelease(version) {
+  return (
+    semver.prerelease(version) &&
+    semver.minor(version) === 0 &&
+    semver.patch(version) === 0
+  );
+}
+
+const baseBranch = isMajorPrerelease(PACKAGE_VERSION)
+  ? `v${semver.major(PACKAGE_VERSION)}`
+  : 'master';
+
+// Polaris bot settings
+const polarisBotName = 'Shopify Polaris Bot';
+const polarisBotEmail = 'shopify-polaris-bot@users.noreply.github.com';
+const polarisBotToken = secrets.github['shopify-polaris'];
+const shopifyPolarisBotGitOverride = `GIT_COMMITTER_NAME='${polarisBotName}' GIT_COMMITTER_EMAIL='${polarisBotEmail}'`;
+
+/**
+ * BulkExecute - Run a bunch of execSync one at a time
+ *
+ * @param {array}  tasks   - List of tasks to run
+ * @param {object} options - Specific options for execSync
+ */
+function BulkExecute(tasks, options) {
+  tasks.forEach((task) => {
+    execSync(task.command, options ? options : {});
+  });
+}
+
+// Each of the repositories to create pull requests updating polaris
+const repositories = ['polaris-styleguide', 'web'];
+
+// Iterate through the repositories
+const tasks = repositories.map((repository) => {
+  // Create a new promise for each repository
+  return new Promise((resolve, reject) => {
+    try {
+      const repositoryDirectory = pathResolve(sandbox, repository);
+
+      // Message that is logged on a failed update
+      const failedUpdateMessage = `It seems your automatic branch creation for ${repository} failed. This can be due to many reasons. To resolve this follow these steps and replace [release-version] with the version you are publishing:
```

Comment:
Can you make this a JS template and move it to the top of the file? This big chunk of text in the middle of the logic makes it hard to understand what's going on

Reply:
Yeah I will do that for both, I will need to pass the repository into the string.